development-process.md

```diff
@@ -0,0 +1,144 @@
+- PR is considered "approved" when you get a +1 (thumbs up) from the assigned reviewer
+
+#### &#10146; (As necessary) Rebase onto master and resolve conflicts when they occur
+
+As you work on your PR, changes may be made to master that cause would-be merge conflicts. The
+merge-ability of your branch will be indicated near the bottom of your PR page. If conflicts are detected,
+you should rebase your branch onto master, fix the conflicts, and push your branch. Here is a [rebasing
+guide maintained by edX](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request). The
+process looks something like this in the command line:
+
+    # Assuming (a) your issue branch is currently checked out...
+    git fetch && git rebase master
+    # Resolve merge conflicts, commit, repeat until done
+    # When merge conflicts are all resolved...
+    git push
+
+That message near the bottom of your PR page should have a green check mark and read "This branch has no
+conflicts with the base branch" when you have successfully resolved and pushed.
+
+
+### Step 3) Merging
+
+#### &#10146; "Squash" the commits on your branch
+
+You may have many incremental commits on your issue branch. In order to keep the git log for the master
+branch clean and readable, we "squash" all issue branch commits into one commit before merging. The commit
+message should be a past tense description of your entire PR, eg: "Updated the home page to match new design".
+
+The actual squashing can be done in a few different ways. The two main approaches are (1) interactive rebase,
+and (2) soft reset. The aforementioned edX rebasing guide contains information about
+[squashing branch commits with an interactive rebase](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#squash-your-changes).
+This is the recommended method, but you can also try accomplishing this with a soft reset:
+
+    git fetch
+    git merge-base CURRENT_BRANCH master
+    # 'merge-base will' give you a commit hash. Copy that value for this next command
+    git reset --soft COMMIT_HASH
+    git commit -am YOUR_COMMIT_MESSAGE
+    # The above will cause your branch to diverge, so you'll need force-push.
+    # Make sure your git 'push.default' config is set to 'simple' before running this.
+    git push -f
+
+When this is done, your git log should show one commit for all the changes in your branch followed by the
+'merge base' of your branch.
+
+#### &#10146; Merge your PR into master via Github
+
+We merge PR's by button-click in Github for a few reasons. One of them to avoid possible human error in
+pushing to master in the command line. Near the bottom of your PR page, you should see a "Merge pull request"
+button. Click that.
+
+#### &#10146; Close your branch and move your issue
+
+You should see an option to close your branch after you merge in Github. When the branch is closed, go to
+the "Boards" tab in your repo and move your issue over to "Done".
```

Comment:
The issue should get moved to "closed" automatically if the "fixed #123" style of commit message is used.